layer/layer_store.go

```diff
@@ -473,17 +410,12 @@ func (ls *layerStore) Mount(name string, parent ChainID, mountLabel string, init
 		}()
 	}
 
-	mountID := name
-	if runtime.GOOS != "windows" {
-		// windows has issues if container ID doesn't match mount ID
-		mountID = stringid.GenerateRandomID()
-	}
-
 	m = &mountedLayer{
 		name:       name,
 		parent:     p,
-		mountID:    mountID,
+		mountID:    ls.mountID(name),
 		layerStore: ls,
+		references: map[RWLayer]*referencedRWLayer{},
```

Comment:
That's bit of a strage construction where `key == value`. Maybe a slice would be enough.